completions/Makefile.am

```diff
@@ -503,6 +502,11 @@ bashcomp_DATA = 2to3 \
 		zopfli \
 		zopflipng
 
+
+if SOLARIS
+bashcomp_DATA += pkgutil
+endif
```

Comment:
Looks like the following utilities would benefit from a similar treatment: freebsd-update, killall, kldload, kldunload, portsnap, route, and watch.

We don't necessarily have to address them here, but I would not object to doing so at all. But if not, we should make a followup PR for them once this is in.

Aside, I don't know if we should make the SOLARIS etc defines more specific to avoid clashes with some possibly existing ones or if it's no concern at all, e.g. BASHCOMP_OS_SOLARIS, BASHCOMP_OS_LINUX, etc.

Reply:
The defines should be fine -- AM_CONDITIONAL defines live in their own namespace and this is generated as SOLARIS_TRUE / SOLARIS_FALSE in ac_subst_vars. In general, you are supposed to be allowed to choose your own arbitrary project-specific names here.

Good point on the other freebsd programs: https://gitweb.gentoo.org/repo/gentoo.git/tree/app-shells/bash-completion/bash-completion-2.14.0.ebuild?id=8aa148634191c504242b4c4ab82f2ef622aede22

Gentoo is already opportunistically deleting a bunch of freebsd stuff for similar reasons.

Reply:
- Can we also switch the files for `completions/_umount{,.linux}` and `completions/_mount{,.linux}` on the installation stage?
- Also, `route`, `watch`, and `killall` have similar `OSTYPE` checks.
- `ifup` seems to depend on a condition based on the output of `uname -s`.
- `_modules` also seem to depend on the system arrangement, but I'm not sure if we should skip the installation because the condition can be dynamic. The required directory might be added later.

Reply:
"can" we is a good question.

> We don't necessarily have to address them here, but I would not object to doing so at all. But if not, we should make a followup PR for them once this is in.

I already dug deeper and handled everything that was simple, clean and obvious to me. Some of those files, I'm not at all sure (yet) how they should be handled, and the rest have OS conditionals which I don't quite understand but feel a bit wrong (e.g. there are comments on other tickets about how the checks are being used as a shortcut for "probably not the same command with the same options").

The current changes I made, I'm highly confident they are always correct, so I would rather merge them as is and follow up on the rest.

Reply:
Thanks for the reply. I understand that you won't work on it within this PR.

> the rest have OS conditionals which I don't quite understand but feel a bit wrong (e.g. there are comments on other tickets about how the checks are being used as a shortcut for "probably not the same command with the same options").

Which one are you talking about? I found #453, but the explanation about the check there appears understandable to me. What do you feel a bit wrong with? Are there other tickets related to this?

The conditions for `ifup` and `_modules` would be hard to check for cross-compiling, so we shouldn't try to switch their installation probably.

The other tests using `OSTYPE` seem consistent with the checks of the completions for which you adjusted the installation. You may think it is unneeded because it sets up completions in Gentoo Linux as expected, but it won't set up completions in non-Linux systems. I currently believe it would be consistent to switch the installations of the completions for Linux in a similar way as you did for non-Linux completions.